openedx/features/content_tagging/rest_api/v1/tests/test_views.py

```diff
@@ -0,0 +1,964 @@
+        ("userA", "tC1", status.HTTP_200_OK),
+        ("userA", "tC2", status.HTTP_200_OK),
+        ("userA", "tB1", status.HTTP_403_FORBIDDEN),
+        ("userA", "tB2", status.HTTP_404_NOT_FOUND),
+        ("userS", "tA2", status.HTTP_200_OK),
+        ("userS", "tB2", status.HTTP_200_OK),
+        ("userS", "tC2", status.HTTP_200_OK),
+        ("userS", "t1", status.HTTP_200_OK),
+        ("userS", "t2", status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_patch_taxonomy(self, user_attr, taxonomy_attr, expected_status):
+        taxonomy = getattr(self, taxonomy_attr)
+
+        url = TAXONOMY_ORG_DETAIL_URL.format(pk=taxonomy.pk)
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.patch(url, {"name": "new name"}, format="json")
+        assert response.status_code == expected_status
+
+        # If we were able to patch the taxonomy, check if the name changed
+        if status.is_success(expected_status):
+            response = self.client.get(url)
+            check_taxonomy(
+                response.data,
+                response.data["id"],
+                **{
+                    "name": "new name",
+                    "description": taxonomy.description,
+                    "enabled": taxonomy.enabled,
+                    "required": taxonomy.required,
+                },
+            )
+
+    @ddt.data(
+        (False, status.HTTP_403_FORBIDDEN),
+        (True, status.HTTP_403_FORBIDDEN),
+    )
+    @ddt.unpack
+    def test_patch_taxonomy_system_defined(self, update_value, expected_status):
+        """
+        Test that we can't patch system_defined field
+        """
+        url = TAXONOMY_ORG_DETAIL_URL.format(pk=self.st1.pk)
+
+        self.client.force_authenticate(user=self.userS)
+        response = self.client.patch(
+            url, {"name": "new name", "system_defined": update_value}, format="json"
+        )
+        assert response.status_code == expected_status
+
+        # Verify that system_defined has not changed
+        response = self.client.get(url)
+        assert response.data["system_defined"] is True
+
+    @ddt.data(
+        (None, "t1", status.HTTP_403_FORBIDDEN),
+        ("user", "t1", status.HTTP_403_FORBIDDEN),
+        ("user", "tA1", status.HTTP_403_FORBIDDEN),
+        ("userA", "tA1", status.HTTP_204_NO_CONTENT),
+        ("userA", "tA2", status.HTTP_204_NO_CONTENT),
+        ("userA", "tC1", status.HTTP_204_NO_CONTENT),
+        ("userA", "tC2", status.HTTP_204_NO_CONTENT),
+        ("userA", "tB1", status.HTTP_403_FORBIDDEN),
+        ("userA", "tB2", status.HTTP_404_NOT_FOUND),
+        ("userS", "tA2", status.HTTP_204_NO_CONTENT),
+        ("userS", "tB2", status.HTTP_204_NO_CONTENT),
+        ("userS", "tC2", status.HTTP_204_NO_CONTENT),
+        ("userS", "t1", status.HTTP_204_NO_CONTENT),
+        ("userS", "t2", status.HTTP_204_NO_CONTENT),
```

Comment:
Can you add checks for deleting the system taxonomy too, just in case?
```suggestion
        ("userS", "t2", status.HTTP_204_NO_CONTENT),
        ("userS", "st1", status.HTTP_403_FORBIDDEN),
        ("userS", "st2", status.HTTP_403_FORBIDDEN),
```